js/blockedLinks.js

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* globals gPinnedLinks, gUserDatabase */
+"use strict";
+
+/**
+ * Singleton that keeps track of all blocked links in the grid.
+ */
+(function(exports, userDB) {
+  const gBlockedLinks = {
+    /**
+     * The cached list of blocked links.
+     */
+    _links: new Set(),
+
+
+    /**
+     * Load the blocked links from userDB and cache them.
+     */
+    init() {
+      var loadPromise = userDB.load("prefs", "blockedLinks");
+      loadPromise.then(loadedLinks => {
+        if (loadedLinks && loadedLinks.length) {
+          this._links = new Set(JSON.parse(loadedLinks));
```

Comment:
Is add able to handle arrays? I created a new set here because the constructor will convert an iterable (`loadedLinks`) into a set